common/utils.go

```diff
@@ -83,3 +116,136 @@ func FindRockspec(path string) (string, error) {
 
 	return "", nil
 }
+
+// GetHomeDir returns current home directory
+func GetHomeDir() (string, error) {
+	usr, err := user.Current()
+	if err != nil {
+		return "", err
+	}
+	return usr.HomeDir, nil
+}
+
+// RandomString generates random string length n
+func RandomString(n int) string {
+	var letters = []rune("abcdefghijklmnopqrstuvwxyz0123456789")
+
+	s := make([]rune, n)
+	for i := range s {
+		s[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(s)
+}
+
+// IsSubDir checks if directory is subdirectory of other
+func IsSubDir(subdir string, dir string) (bool, error) {
+	subdirAbs, err := filepath.Abs(subdir)
+	if err != nil {
+		return false, err
+	}
+
+	dirAbs, err := filepath.Abs(dir)
+	if err != nil {
+		return false, err
+	}
+
+	if dirAbs == subdirAbs {
+		return true, nil
+	}
+
+	return strings.HasPrefix(subdirAbs, fmt.Sprintf("%s/", dirAbs)), nil
+}
+
+// ClearDir removes all files from specified directory
+func ClearDir(dirPath string) error {
+	files, err := filepath.Glob(filepath.Join(dirPath, "*"))
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		err = os.RemoveAll(file)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GitIsInstalled checks if git binary is in the PATH
+func GitIsInstalled() bool {
+	_, err := exec.LookPath("git")
+	return err == nil
+}
+
+// IsGitProject checks if specified path is a git project
+func IsGitProject(path string) bool {
+	fileInfo, err := os.Stat(filepath.Join(path, ".git"))
+	return err == nil && fileInfo.IsDir()
+}
+
+// HasPerm checks if specified file has permissions
+func HasPerm(fileInfo os.FileInfo, perm os.FileMode) bool {
```

Comment:
HasPermissions?

Reply:
`perm` word is used in the `os` module doc, I've decided that it's OK.

Reply:
Ok